src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,159 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public boolean add(T value) {
-        return false;
+        Node<T> node = new Node<>(tail, value, null);
+        addHeadOrTail(node);
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndex(index, size);
+        if (index == size) {
+            add(value);
+        } else {
+            addByIndex(value, index);
+        }
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (T value : list) {
+            add(value);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index, size - 1);
+        return findNode(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index, size - 1);
+        Node<T> removeNode = findNode(index);
+        T removeValue = removeNode.item;
+        removeNode.item = value;
+        return removeValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index, size - 1);
+        return unlink(index);
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> currentNode = head;
+        for (int i = 0; i < size; i++) {
+            if (object == currentNode.item || (object != null && object.equals(currentNode.item))) {
+                unlink(i);
+                return true;
+            } else {
```

Comment:
do we need else block here?

Reply:
No, because when we find the right node, we will finish the method